packages/pieces/community/pinterest/src/lib/actions/create-board.ts

```diff
@@ -0,0 +1,50 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import { makeRequest } from '../common';
+import { pinterestAuth } from '../common/auth';
+import { HttpMethod } from '@activepieces/pieces-common';
+
+export const createBoard = createAction({
+  auth: pinterestAuth,
+  name: 'createBoard',
+  displayName: 'Create Board',
+  description: 'Create a new Pinterest board for organizing Pins.',
+  props: {
+    name: Property.ShortText({
+      displayName: 'Board Name',
+      required: true,
+      description: 'The name of the board.'
+    }),
+    description: Property.LongText({
+      displayName: 'Description',
+      required: false,
+      description: 'The description of the board.'
+    }),
+    privacy: Property.StaticDropdown({
+      displayName: 'Privacy',
+      required: true,
+      options: {
+        options: [
+          { label: 'Public', value: 'PUBLIC' },
+          { label: 'Secret', value: 'SECRET' }
+        ]
+      },
+      description: 'Set the board as public or secret.'
+    }),
+    is_ads_only: Property.Checkbox({
+      displayName: 'is_ads_only',
+      description: 'If set to true, the board will be ad-only and can store ad-only Pins.',
+      defaultValue: false,
+      required: false
+    })
+  },
+  async run({ auth, propsValue }) {
```

Comment:
Missing Query Parameter Handling: In the run function, you need to handle the ad_account_id parameter:
```suggestion
     async run({ auth, propsValue }) {
     const { name, description, privacy, is_ads_only, ad_account_id } = propsValue;
     const body: any = {
       name,
       is_ads_only
     };
     if (description) body.description = description;
     if (privacy) body.privacy = privacy; // Only include if specified
     
     let path = '/boards';
     if (ad_account_id) {
       path = `/boards?ad_account_id=${ad_account_id}`;
     }
     
     return await makeRequest(auth.access_token as string, HttpMethod.POST, path, body);
   }
```